irohad/main/impl/peers_file_reader_impl.cpp

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "peers_file_reader_impl.hpp"
+
+#include <fstream>
+
+#include "cryptography/public_key.hpp"
+#include "interfaces/common_objects/types.hpp"
+#include "parser/parser.hpp"
+
+using namespace iroha::main;
+
+boost::optional<std::string> PeersFileReaderImpl::openFile(
+    const std::string &name) {
+  std::ifstream file(name);
+  if (not file) {
+    return boost::none;
+  }
+
+  std::string str((std::istreambuf_iterator<char>(file)),
+                  std::istreambuf_iterator<char>());
+  return str;
+}
+
+boost::optional<std::vector<std::unique_ptr<shared_model::interface::Peer>>>
+PeersFileReaderImpl::readPeers(
+    const std::string &peers_data,
+    std::shared_ptr<shared_model::interface::CommonObjectsFactory>
+        common_objects_factory) {
+  auto strings = parser::split(peers_data);
```

Comment:
I suggest to trim (or strip in python) the input data before parser::split.

In other words - remove all the spaces and space-like characters from the beginning and from the end of the input data.

Reply:
Don't worry about triming since it would be implicitly done by `istringstream`.

Reply:
I think we should stop worrying about this kind of problems and delegate it to standard format parsers like CSV or JSON.